micrometer-api/src/main/java/io/micrometer/api/instrument/MeterRegistry.java

```diff
@@ -85,6 +88,31 @@
     private final Config config = new Config();
     private final More more = new More();
 
+    private final ThreadLocal<Observation> localObservation = new ThreadLocal<>();
+
+    private final ObservationConfig observationConfig = new ObservationConfig();
+
+    @Nullable
+    @Override
+    public Observation getCurrentObservation() {
+        return this.localObservation.get();
+    }
+
+    @Override
+    public void setCurrentObservation(@Nullable Observation current) {
+        this.localObservation.set(current);
+    }
+
+    @Override
+    public ObservationConfig observationConfig() {
+        return this.observationConfig;
+    }
+
+    //TODO Want this under observationConfig but not sure that's possible
+    public void withTimerObservationHandler() {
```

Comment:
Ideally, yes. That was my intention, but I didn't see a practical way to do that without making it a method on `ObservationConfig`, which shouldn't directly know about `Timer`. I couldn't think of a way to add that only in `MeterRegistry`. We should definitely think of something better than what's there now. It was temporary since it was late and I was out of ideas but just wanted to demonstrate the general intention.

Reply:
To achieve what you mentioned, it works well enough to just return `observationConfig` after the current logic, but what I meant was that ideally the method itself would be nested under `MeterRegistry#observationConfig()` instead of a top-level method on `MeterRegistry`, I think. But that doesn't feel possible. So the question is perhaps are we okay with this (awkwardly) being a top-level method on MeterRegistry or perhaps under `Config`? If so, we can update it as you suggested.